src/constants.mjs

```diff
@@ -71,11 +78,13 @@ export const DOC_API_HEADING_TYPES = [
   },
   {
     type: 'classMethod',
-    regex: /^Static method: +`?([A-Z]\w+(?:\.[A-Z]\w+)*\.\w+)\([^)]*\)`?$/i,
+    regex:
+      /^Static method: +`?[A-Z]\w+(?:\.[A-Z]\w+)*(?:(\[\w+\.\w+\])|\.(\w+))\([^)]*\)`?$/i,
   },
   {
     type: 'property',
-    regex: /^(?:Class property: +)?`?([A-Z]\w+(?:\.[A-Z]\w+)*\.\w+)`?$/i,
+    regex:
+      /^(?:Class property: +)?`?[A-Z]\w+(?:\.[A-Z]\w+)*(?:(\[\w+\.\w+\])|\.(\w+))`?$/i,
```

Comment:
This is still vulnerable to ReDos

Reply:
How do you know? I don't think it is.

Reply:
There are tools that check a Regex against Redos.

Reply:
I know. I checked it against https://devina.io/redos-checker, https://makenowjust-labs.github.io/recheck/playground/

Reply:
That's funny, I checked on both:

- https://devina.io/redos-checker
- https://redosdetector.com/?pattern=&caseInsensitive=false&unicode=false

And the 2nd one gives me:

![image](https://github.com/user-attachments/assets/dfa20932-cf66-4ea9-9fb9-f3fcfaf66ee3)

But the other tool gives me:

![image](https://github.com/user-attachments/assets/b910ce19-c9d3-4727-83ba-a9534a5d4148)

Reply:
It should be fine. I ran it against an automated test (recheck), and it took ~0.7s to complete in a worst-case scenario